pkg/util/fixtures/populate/populate.go

```diff
@@ -0,0 +1,145 @@
+		Primes: []*big.Int{
+			fromBase10("130903255182996722426771613606077755295583329135067340152947172868415809027537376306193179624298874215608270802054347609836776473930072411958753044562214537013874103802006369634761074377213995983876788718033850153719421695468704276694983032644416930879093914927146648402139231293035971427838068945045019075433"),
+			fromBase10("109348945610485453577574767652527472924289229538286649661240938988020367005475727988253438647560958573506159449538793540472829815903949343191091817779240101054552748665267574271163617694640513549693841337820602726596756351006149518830932261246698766355347898158548465400674856021497190430791824869615170301029"),
+		},
+	}
+	sample2048Key.Precompute()
+}
+
+// Walk is a recursive struct value population function. Given a pointer to an arbitrarily complex value v, it fills
+// in the complete structure of that value, labelling each element with the path by which it was found.
+//
+// This function has the following caveats:
+//  - Signed integers are set to int(3)
+//  - Unsigned integers are set to uint(3)
+//  - Floating point numbers are set to float(3.0)
+//  - Booleans are set to True
+//  - Arrays and slices are allocated 3 elements
+//  - Maps are allocated 3 elements
+//  - Only map[string][string] types are supported
+//  - strings are set to the value of the path taken to reach the string
+func Walk(v interface{}) {
+	val := reflect.ValueOf(v)
+	if val.Kind() != reflect.Ptr {
+		panic("argument is not a pointer to a value")
+	}
+	walk(val, "")
+}
+
+// walk fills in the complete structure of a complex value v using path as the root of the labelling.
+func walk(v reflect.Value, path string) {
+	switch v.Kind() {
+	case reflect.Ptr:
+		if v.IsNil() {
+			v.Set(reflect.New(v.Type().Elem()))
+		}
+		walk(v.Elem(), path)
+	case reflect.Struct:
+		// do not go on with the recursion if it isn't one of the core openshift-azure types
+		if !strings.HasPrefix(v.Type().PkgPath(), "github.com/openshift/openshift-azure/") ||
+			strings.HasPrefix(v.Type().PkgPath(), "github.com/openshift/openshift-azure/vendor/") {
+			return
+		}
+		for i := 0; i < v.NumField(); i++ {
+			field := v.Field(i)
+			if field.IsValid() {
+				newpath := extendPath(path, v.Type().Field(i).Name, v.Kind())
+				switch field.Interface().(type) {
+				case *rsa.PrivateKey:
+					field.Set(reflect.ValueOf(sample2048Key))
+				}
+				walk(field, newpath)
+			}
+		}
+	case reflect.Array, reflect.Slice:
+		// if the array/slice has length 0 create a new slice of length 1
+		if v.Len() == 0 {
+			v.Set(reflect.MakeSlice(v.Type(), 3, 3))
```

Comment:
the following avoids a very edge case on a zero length array:

```golang
case reflect.Slice:
	if v.Len() == 0 {
		v.Set(reflect.MakeSlice(v.Type(), 1, 1))
	}
	fallthrough

case reflect.Array:
	for i := 0; i < v.Len(); i++ {
...
```

prefer slice length of 1 instead of 3